admiral/pkg/clusters/serviceentry.go

```diff
@@ -160,7 +160,11 @@ func modifyServiceEntryForNewServiceOrPod(
 		sourceClusters                        []string
 		isAdditionalEndpointGenerationEnabled bool
 		deployRolloutMigration                = make(map[string]bool)
-		sourceIngressVirtualService           = make(map[string]*v1alpha3.VirtualService)
+
+		// Holds the ingress virtualservices for the source cluster
+		sourceIngressVirtualService = make(map[string]*v1alpha3.VirtualService)
+		// Holds the VS destinations for the TLSRoutes
+		sourceClusterToDestinations = make(map[string]map[string][]*networking.RouteDestination)
```

Comment:
can this be generated based on `sourceIngressVirtualService`
Will this add memory pressure on the normal operation of Admiral?

Reply:
@adilfulara , it should not add memory pressure because we are using pointers where possible. 
I can take a look if they can be combined, I just followed the same logic as SE